bungee/src/main/java/us/myles/ViaVersion/bungee/service/ProtocolDetectorService.java

```diff
@@ -27,16 +27,19 @@ public ProtocolDetectorService(BungeePlugin plugin) {
     public static Integer getProtocolId(String serverName) {
         // Step 1. Check Config
         Map<String, Integer> servers = ((BungeeConfigAPI) Via.getConfig()).getBungeeServerProtocols();
-        if (servers.containsKey(serverName)) {
-            return servers.get(serverName);
+        Integer protocol = servers.get(serverName);
+        if (protocol != null) {
+            return protocol;
         }
         // Step 2. Check Detected
-        if (detectedProtocolIds.containsKey(serverName)) {
-            return detectedProtocolIds.get(serverName);
+        Integer detectedProtocol = detectedProtocolIds.get(serverName);
+        if (detectedProtocol != null) {
+            return detectedProtocol;
         }
         // Step 3. Use Default
-        if (servers.containsKey("default")) {
-            return servers.get("default");
+        Integer defaultProtocol = servers.get("default");
+        if (defaultProtocol != null) {
+            return detectedProtocol;
```

Comment:
This should probably return the default protocol?

Reply:
My bad, would have completely missed it.. Thanks!